hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/TenantConfigurationDashboardView.java

```diff
@@ -74,76 +49,38 @@
+    private final List<Binder<? extends ProxySystemConfigWindow>> binders = Lists.newArrayList();
 
     @Autowired
     TenantConfigurationDashboardView(final VaadinMessageSource i18n, final UiProperties uiProperties,
-            final UINotification uINotification, final SystemManagement systemManagement,
-            final DistributionSetTypeManagement distributionSetTypeManagement,
-            final TenantConfigurationManagement tenantConfigurationManagement,
-            final SecurityTokenGenerator securityTokenGenerator,
-            final ControllerPollProperties controllerPollProperties, final SpPermissionChecker permChecker) {
+            final UINotification uINotification, final List<BaseConfigurationView<? extends ProxySystemConfigWindow>> configurationViews) {
         this.i18n = i18n;
         this.uiProperties = uiProperties;
         this.uINotification = uINotification;
-        this.systemManagement = systemManagement;
-        this.tenantConfigurationManagement = tenantConfigurationManagement;
-        this.securityTokenGenerator = securityTokenGenerator;
-
-        this.binder = new Binder<>();
-        binder.setBean(populateAndGetSystemConfig());
-
-        this.defaultDistributionSetTypeLayout = new DefaultDistributionSetTypeLayout(systemManagement, i18n,
-                permChecker, distributionSetTypeManagement, binder);
-
-        this.authenticationConfigurationView = new AuthenticationConfigurationView(i18n, uiProperties,
-                securityTokenGenerator, binder);
-        this.pollingConfigurationView = new PollingConfigurationView(i18n, controllerPollProperties,
-                tenantConfigurationManagement, binder);
-        this.repositoryConfigurationView = new RepositoryConfigurationView(i18n, uiProperties, binder);
-        this.rolloutConfigurationView = new RolloutConfigurationView(i18n, uiProperties, binder);
+        this.autowiredConfigurationViews = configurationViews;
     }
 
     /**
      * Init method adds all Configuration Views to the list of Views.
      */
     @PostConstruct
     public void init() {
-        if (defaultDistributionSetTypeLayout.getComponentCount() > 0) {
-            customComponents.add(defaultDistributionSetTypeLayout);
-        }
-        customComponents.add(repositoryConfigurationView);
-        customComponents.add(rolloutConfigurationView);
-        customComponents.add(authenticationConfigurationView);
-        customComponents.add(pollingConfigurationView);
-        if (customConfigurationViews != null) {
-            configurationViews.addAll(
-                    customConfigurationViews.stream().filter(ConfigurationGroup::show).collect(Collectors.toList()));
-        }
+        // filter the autowired views
+        filteredConfigurationViews.addAll(autowiredConfigurationViews.stream().filter(c -> c.getComponentCount() > 0)
+                .filter(ConfigurationGroup::show).collect(Collectors.toList()));
```

Comment:
```suggestion
        filteredConfigurationViews = autowiredConfigurationViews.stream().filter(c -> c.getComponentCount() > 0)
                .filter(ConfigurationGroup::show).collect(Collectors.toList()));
```
No need to first initialize a ArrayList for `filteredConfigurationViews`

Reply:
Done. Changes applied as suggested